spec/models/product_spec.rb

```diff
@@ -1,7 +1,15 @@
 require "rails_helper"
 
-RSpec.describe Product do
+RSpec.describe Product, :with_opensearch do
```

Comment:
This will significantly slow down the specs so would suggest removal and only adding to those examples where we are testing search functionality.

```suggestion
RSpec.describe Product do
```

Reply:
I've added this because the instance needs to be created in order to have an `id`, and creation indexes the model with opensearch. Is there an alternative approach?

Reply:
This appears to disrupt `it_behaves_like "a batched search model"` in the example above unfortunately, this shared example aready includes `:with_opensearch`...

```
1) Product behaves like a batched search model #search_in_batches when searching in batches returns the expected objects
   Failure/Error: described_class.import refresh: :wait_for

   NoMethodError:
     private method `select' called for nil:NilClass
   Shared Example Group: "a batched search model" called from ./spec/models/product_spec.rb:4
```

Reply:
Ah OK, I didn't realise `with_opensearch` was already included there and there weren't any other examples in this spec.

In that case, no significant performance deficit introduced, but ideally for a slightly faster run and to avoid slowing down any future examples we add to this spec, I suggest we could refactor like so:

```ruby
  describe "#psd_ref" do
    let(:id) { 123 }
    let(:product) { build :product, id: id }

    it "returns a reference formed with 'psd-' and the product's ID" do
      expect(product.psd_ref).to eq("psd-#{id}")
    end
  end
```

Reply:
Perfect, I'll switch to building but not saving 👍